opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -1216,45 +1215,23 @@ typedef Eigen::Array<double,
             }
         }
 
-        //const V sumSat = sw + so + sg;
-        //sw = sw / sumSat;
-        //so = so / sumSat;
-        //sg = sg / sumSat;
-
-        // Update the reservoir_state
-        if (active_[Water]) {
-            for (int c = 0; c < nc; ++c) {
-                reservoir_state.saturation()[c*np + pu.phase_pos[ Water ]] = sw[c];
-            }
-        }
-
-        if (active_[Gas]) {
-            for (int c = 0; c < nc; ++c) {
-                reservoir_state.saturation()[c*np + pu.phase_pos[ Gas ]] = sg[c];
-            }
-        }
-
-        if (active_[ Oil ]) {
-            for (int c = 0; c < nc; ++c) {
-                reservoir_state.saturation()[c*np + pu.phase_pos[ Oil ]] = so[c];
-            }
-        }
-
         // Update rs and rv
         const double drmaxrel = drMaxRel();
         V rs;
         if (has_disgas_) {
             const V rs_old = Eigen::Map<const V>(&reservoir_state.gasoilratio()[0], nc);
             const V drs = isRs_ * dxvar;
-            const V drs_limited = sign(drs) * drs.abs().min(rs_old.abs()*drmaxrel);
+            const V drs_limited = sign(drs) * drs.abs().min( (rs_old.abs()*drmaxrel).max( ones*1.0));
```

Comment:
This change seems to be to always allow a change of at least 1, which is probably necessary if rs starts out very low (near zero), and the similar change below seems to always allow a change of 1e-3.

Is there any particular reasons for choosing these numbers or are they just "reasonable values that work" (and are quite a bit lower than expected normal values for rs/rv)?